src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -100,5 +100,53 @@ public static async Task ReadValueTypeCollectionPassingNullValueAsync()
                 Assert.Null(valueTypeCollection);
             }
         }
+
+        public class TestDataGenerator : TheoryData<byte[], JsonSerializerOptions, ulong>
+        {
+            public TestDataGenerator()
+            {
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, default, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 1 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 2 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 3 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 4 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 15 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49 }, new JsonSerializerOptions { DefaultBufferSize = 15 }, 1111111111111111111);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 16 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49 }, new JsonSerializerOptions { DefaultBufferSize = 16 }, 1111111111111111111);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49 }, new JsonSerializerOptions { DefaultBufferSize = 17 }, 1);
+                Add(new byte[] { 0xEF, 0xBB, 0xBF, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49 }, new JsonSerializerOptions { DefaultBufferSize = 17 }, 1111111111111111111);
+            }
+        }
+
+        [Theory]
+        [ClassData(typeof(TestDataGenerator))]
```

Comment:
This is interesting. I have never used this before.

Usually I just use `MemberData` and pass in an `IEnumerable<object[]>`. Any particular reason to go with this approach?

Example:
https://github.com/dotnet/corefx/blob/42ecf717478504fd7cc072f3586c6edd1ba57daa/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L3620-L3630

Looks like ClassData is used primarily in `System.Linq.Expressions` tests, whereas `MemberData` is used quite a lot more.
https://github.com/dotnet/corefx/search?q=ClassData&unscoped_q=ClassData
https://github.com/dotnet/corefx/search?q=MemberData&unscoped_q=MemberData

nit: If there isn't a particular reason for `ClassData`, let's use `MemberData` and avoid creating the `TestDataGenerator` class.

Reply:
I changed it to use MemberData. The only reason I use it is because ClassData enables strong typing.
https://andrewlock.net/creating-strongly-typed-xunit-theory-test-data-with-theorydata/